### ISSUE TITLE
New "event constructing steps" could use an example

### ISSUE THREAD

**Issue by marcoscaceres** [whatwg/dom#683]
The new "event constructing steps" could use an example, because they are quite circular and somewhat hard to follow. 

The problem is that you can have a normal user-land constructor (e.g., developer creates an untrusted event named "e" for whatever reason) and the UA itself runs the "event constructing steps", which are internal (but set the event to some specific name).

Another thing that should be clarified is where does "constructor" play a role in the "event constructing steps" (if any).

**Comment by marcoscaceres** [comment]
Actually, ignore the whole type/name thing - this was a problem with my own spec. But an example would be good nonetheless :) 

I think a good example that goes from `constructor` into calling  "event constructing steps" would be good.

**Comment by annevk** [comment]
Who is the audience of this example?

**Comment by saschanaz** [comment]
For anyone finding an example, there is a real world one here: https://w3c.github.io/payment-request/#merchantvalidationevent-constructor
